trollflow2/launcher.py

```diff
@@ -438,3 +440,61 @@ def sendmail(config, trace):
     pid = Popen([sendmail, "-t", "-oi"], stdin=PIPE)
     pid.communicate(msg.as_bytes())
     pid.terminate()
+
+
+def launch(args_in):
+    """Launch the processing."""
+    args = parse_args(args_in)
+
+    log_config = args.pop("log_config", None)
+    if log_config is not None:
+        with open(log_config) as fd:
+            log_config = yaml.safe_load(fd.read())
+
+    logger = logging.getLogger("satpy_launcher")
+
+    log_queue = Manager().Queue()
+
+    with logging_on(log_queue, log_config):
+        logger.warning("Launching Satpy-based runner.")
+        product_list = args.pop("product_list")
+        test_message = args.pop("test_message")
+        threaded = args.pop("threaded")
+        connection_parameters = args
+
+        runner = Runner(product_list, log_queue, connection_parameters, test_message, threaded)
+        runner.run()
+
+
+def parse_args(args_in):
```

Comment:
you could make `args_in` optional so that parse_args would get sys.argv by itself I think.

Reply:
Would that be an improvement?  It would be harder to test.

Reply:
Yeah, I don't want to make testing any more complex than it already is :sweat_smile: